addons/mail/static/src/js/utils.js

```diff
@@ -53,10 +58,17 @@ function linkify(text, attrs) {
     attrs = _.map(attrs, function (value, key) {
         return key + '="' + _.escape(value) + '"';
     }).join(' ');
-    return text.replace(urlRegexp, function (url) {
-        var href = (!/^https?:\/\//i.test(url)) ? "http://" + url : url;
-        return '<a ' + attrs + ' href="' + href + '">' + url + '</a>';
-    });
+    var curIndex = 0;
+    var result = "";
+    var match;
+    while ((match = urlRegexp.exec(text)) !== null) {
+        result += escapeEntities(text.slice(curIndex, match.index));
+        var url = match[0];
+        var href = (!/^https?:\/\//i.test(url)) ? "http://" + _.escape(url) : _.escape(url);
+        result += '<a ' + attrs + ' href="' + href + '">' + escapeEntities(url) + '</a>';
```

Comment:
I believe `href` attribute should just escape `"` and nothing else? And if we consider `"` is not part of URL anyway, the escape for the `href` part is useless?
Removing it might improve performance too?